### PR TITLE
Final project - Budget-app

### PR DESCRIPTION
**Pull request by benja27**
Pull Request Description
🚀 This pull request implements the following requirements in the application:

Interactions
Splash Screen 🌟

Created a welcome screen with the application name and links to the registration and login pages.
Sign Up and Log In Pages 📝

Users can register in the application by providing their full name, email, and password (all mandatory).
Users can log in to the application using their email and password.
Users cannot access pages that require authentication if they are not logged in.
Home Page (Categories Page) 🏡

When users log in, they are presented with the categories page.
For each category, users can see its name, icon, and the total amount of all transactions belonging to that category.
Clicking on a category item navigates to the transactions page for that category.
Added a "Add a New Category" button at the bottom, which takes users to the page to create a new category.
Transactions Page 💰

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page, users can see the total amount for the category (sum of all amounts of transactions in that category).
Added a "Add a New Transaction" button at the bottom, which takes users to the page to create a new transaction.
Clicking the "Back" button (←) navigates users back to the home page.
"Add a New Category" Page 📁

Users fill out a form to create a new category, indicating their name and icon (both mandatory).
Clicking the "Save" button creates the new category and takes users to the home page on success.
Clicking the "Back" button (←) navigates users back to the home page.
"Add a New Transaction" Page 📄

Users fill out a form to create a new transaction with:
Name (mandatory)
Amount (mandatory)
Categories (at least one mandatory)
Clicking the "Save" button creates the new transaction and takes users to the transactions page for that category.
Clicking the "Back" button (←) navigates users to the transactions page for that category.
Testing Requirements
✅ Created unit and integration tests for all the most important components of the RoR application.

Technical Requirements
🔧 PostgreSQL is used as the database.
🔐 Devise is used for authentication.
🛡️ All user input is validated to ensure the security of the application.

Other Details
🌐 The project has been deployed and is accessible online.
here is the link: https://rails-b6iy.onrender.com/

###here is a video showing the project:
https://www.loom.com/share/0dcca3000c3c41fb9f0cce5d002069a3

### CODE REVIEW

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @benja27, 


Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights :green_circle: 

- Great job implementing additional delete functionality for each transaction :+1: 
- Fantastic job validating all user's input :+1: 


## Suggested changes

- I'm going to leave some comments here as general review since you didn't push much files to your repo :ok: 

- [x] Based on the requirements of this project, the video presentation should contains a highlights of the piece of code you finds interesting or proud of making. Let's kindly point out this piece of code.
- [x] Based on the requirements of this project, the README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). Let's kindly give credit to the author of our designer as requested :+1: 
- [x] Based on the requirements of this project, for a given category, the list of transactions is presented ordered by the most recent. The last created transaction should be found at the top of the list :+1: 
- [x] Three mandatory inputs are required to create a transaction(name, amount, category). It seems you missed out the category input which can be at least one. Remember a transaction can be created for one or more categories at once.
- [x] Please kindly use only necessary routes for this project instead of the default resourceful routes. You didn't implement functionality for updating user's data in the views.
- [x] Some of your tests are failing as you can see on the screenshot below. Please fix it.
Screenshot from 2023-10-01 01-07-13



_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by benja27** [comment]
> o

All changes have been made!

File: spec/integration/integration_spec.rb [hunk 1]
```diff
@@ -12,6 +12,11 @@
...
   end
 
+  it 'allows the user to visit the index page' do
+    visit groups_path
+    expect(page).to have_content('Groups')
+  end
+
   it 'visit the index page' do
     sign_out(user)
     visit '/users/sign_in'
```
Comment (Owusu-Desmond) [inline spec/integration/integration_spec.rb:22]: - [ ] As requested by the previous reviewer, kindly fix the failing test of your integration and request tests :pray: 
  
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @benja27 , :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some changes you still need to work on before you move on to the next project but you are almost there.



 ## Highlights :+1: 
:heavy_check_mark: Good commit history.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Your project is well documented


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As requested previously, per the requirements of this project, the video presentation should contains a highlights of the piece of code you finds interesting or proud of making. But your video you didn't show and explain any code you are proud of making.



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @benja27, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
congrats 2


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**